examples/with-auth/components/session.js

```diff
@@ -0,0 +1,204 @@
+/**
+ * A class to handle signing in and out and caching session data in sessionStore
+ *
+ * Note: We usewindow.XMLHttpRequest() here rather than fetch because fetch() uses
+ * Service Workers and they cannot share cookies with the browser session
+ * yet (!) so if we tried to get or pass the CSRF token it would mismatch.
```

Comment:
@koutsenko Specifically fetch doesn't yet have ability to do HTTP Only cookies (i.e. the kind that can't be accessed by JavaScript, which give protection from XSS from malicious users or adverts inject JS into your code). The workalike that @sedubois suggests above looks like a good drop in replacement though.

sedubois